### PR TITLE
OSHMEM v1.3: adds fetch and set amos and c11 generic

### PR DESCRIPTION
**Pull request by alex-mikheev**
@jladd-mlnx @igor-ivanov 
based on https://github.com/open-mpi/ompi/pull/2122 with changes for the v2.x branch

### CODE REVIEW

**Comment by jsquyres** [comment]
@alex-mikheev Same comments I just made in https://github.com/open-mpi/ompi/pull/2122#issuecomment-249611743

**Comment by jsquyres** [comment]
This is not merged on master yet -- wait until #2122 is merged and tested.

**Comment by jladd-mlnx** [comment]
#2122 has been merged.

File: oshmem/shmem/fortran/shmem_int4_fetch_f.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2013      Mellanox Technologies, Inc.
+ *                         All rights reserved.
+ * Copyright (c) 2013 Cisco Systems, Inc.  All rights reserved.
+ * $COPYRIGHT$
+ *
```
Comment (igor-ivanov) [inline oshmem/shmem/fortran/shmem_int4_fetch_f.c]: @alex-mikheev please update the header

File: oshmem/shmem/fortran/shmem_int4_set_f.c [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2013      Mellanox Technologies, Inc.
+ *                         All rights reserved.
+ * Copyright (c) 2013 Cisco Systems, Inc.  All rights reserved.
+ * $COPYRIGHT$
+ *
```
Comment (igor-ivanov) [inline oshmem/shmem/fortran/shmem_int4_set_f.c]: @alex-mikheev please update the header

File: oshmem/shmem/fortran/shmem_int8_fetch_f.c [hunk 3]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2013      Mellanox Technologies, Inc.
+ *                         All rights reserved.
+ * Copyright (c) 2013 Cisco Systems, Inc.  All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
```
Comment (igor-ivanov) [inline oshmem/shmem/fortran/shmem_int8_fetch_f.c]: @alex-mikheev please update the header

File: oshmem/shmem/fortran/shmem_int8_set_f.c [hunk 4]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2013      Mellanox Technologies, Inc.
+ *                         All rights reserved.
+ * Copyright (c) 2013 Cisco Systems, Inc.  All rights reserved.
+ * $COPYRIGHT$
+ *
```
Comment (igor-ivanov) [inline oshmem/shmem/fortran/shmem_int8_set_f.c]: @alex-mikheev please update the header

File: oshmem/shmem/fortran/shmem_real4_fetch_f.c [hunk 5]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2013      Mellanox Technologies, Inc.
+ *                         All rights reserved.
+ * Copyright (c) 2013 Cisco Systems, Inc.  All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
```
Comment (igor-ivanov) [inline oshmem/shmem/fortran/shmem_real4_fetch_f.c]: @alex-mikheev please update the header

File: oshmem/shmem/fortran/shmem_real4_set_f.c [hunk 6]
```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2013      Mellanox Technologies, Inc.
+ *                         All rights reserved.
+ * Copyright (c) 2013 Cisco Systems, Inc.  All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
```
Comment (igor-ivanov) [inline oshmem/shmem/fortran/shmem_real4_set_f.c]: @alex-mikheev please update the header

File: oshmem/shmem/fortran/shmem_real8_fetch_f.c [hunk 7]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2013      Mellanox Technologies, Inc.
+ *                         All rights reserved.
+ * Copyright (c) 2013 Cisco Systems, Inc.  All rights reserved.
+ * $COPYRIGHT$
+ *
```
Comment (igor-ivanov) [inline oshmem/shmem/fortran/shmem_real8_fetch_f.c]: @alex-mikheev please update the header

File: oshmem/shmem/fortran/shmem_real8_set_f.c [hunk 8]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2013      Mellanox Technologies, Inc.
+ *                         All rights reserved.
+ * Copyright (c) 2013 Cisco Systems, Inc.  All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
```
Comment (igor-ivanov) [inline oshmem/shmem/fortran/shmem_real8_set_f.c]: @alex-mikheev please update the header

**Comment by jsquyres** [comment]
@igor-ivanov @alex-mikheev FWIW: I use a git hook to verify that I've updated my copyrights; it blocks my commits until I've updated the copyrights properly.  In my `.git/hooks` directory, I have a script named `pre-commit` that looks like this:

```sh
#!/bin/sh
#
# Called by "git commit" with no arguments.  The hook should
# exit with non-zero status after issuing an appropriate message if
# it wants to stop the commit.

function die() {
    rc=$1
    shift;
    printf "*** GIT PRE-COMMIT HOOK FAILED ***:\n"
    echo "$@"
    exit $rc
}

if git rev-parse --verify HEAD >/dev/null 2>&1
then
        against=HEAD
else
        # Initial commit: diff against an empty tree object
        against=4b825dc642cb6eb9a060e54bf8d69288fbee4904
fi

# Redirect output to stderr.
exec 1>&2

# If there are whitespace errors, print the offending file names and fail.
echo "=== Checking for whitespace errors..."
git diff-index --check --cached $against -- || die $? "whitespace errors"

# If we have a copyright-checking script, run it
if test -x ./contrib/update-my-copyright.pl; then
    echo "=== Checking copyrights..."
    ./contrib/update-my-copyright.pl --check-only --quiet || die $? "Copyrights are not up to date"
else
    echo "=== No update-my-copyright.pl script -- skipping"
fi
```

**Comment by jsquyres** [comment]
@jladd-mlnx @igor-ivanov @alex-mikheev The copyrights still aren't updated, and this PR now has conflicts.  I suspect it needs to be rebased now that #2095 (the first of the OSHMEM PRs) was merged.

**Comment by alex-mikheev** [comment]
@jladd-mlnx @igor-ivanov please take a look

**Comment by igor-ivanov** [comment]
:+1:

**Comment by jladd-mlnx** [comment]
👍

**Pull request merged**